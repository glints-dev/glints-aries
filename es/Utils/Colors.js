export var PrimaryColor = {
  glintsyellow: '#FFF240',
  glintsred: '#EC272B',
  glintsblue: '#0BAEEC'
};
export var SecondaryColor = {
  actionblue: '#017EB7',
  darkblue: '#132647',
  green: '#93BD49',
  darkgreen: '#008464',
  magenta: '#FF00E5',
  violet: '#9013FE',
  orange: '#F5A623'
};
export var Greyscale = {
  black: '#000000',
  grey: '#777777',
  devilsgrey: '#666666',
  lightgrey: '#C6C6C6',
  softgrey: '#F3F3F3',
  white: '#FFFFFF'
};