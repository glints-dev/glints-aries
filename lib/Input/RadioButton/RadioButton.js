"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

exports.__esModule = true;
exports["default"] = exports.RadioButton = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutPropertiesLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutPropertiesLoose"));

var React = _interopRequireWildcard(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _RadioButtonStyle = require("./RadioButtonStyle");

var _excluded = ["className", "children", "error", "label", "labelProps", "theme", "border", "disabled", "size"];

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

/** The className will be passed to
 * the main radio button container. ...inputProps is passed to the
 * input tag. */
var RadioButton = function RadioButton(_ref) {
  var className = _ref.className,
      children = _ref.children,
      _ref$error = _ref.error,
      error = _ref$error === void 0 ? false : _ref$error,
      label = _ref.label,
      labelProps = _ref.labelProps,
      theme = _ref.theme,
      _ref$border = _ref.border,
      border = _ref$border === void 0 ? false : _ref$border,
      _ref$disabled = _ref.disabled,
      disabled = _ref$disabled === void 0 ? false : _ref$disabled,
      _ref$size = _ref.size,
      size = _ref$size === void 0 ? 'regular' : _ref$size,
      inputProps = (0, _objectWithoutPropertiesLoose2["default"])(_ref, _excluded);
  var content = /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(_RadioButtonStyle.RadioIcon, {
    error: error,
    theme: theme,
    border: border,
    size: size,
    disabled: disabled,
    "data-testid": "radio-icon"
  }), /*#__PURE__*/React.createElement(_RadioButtonStyle.RadioLabel, {
    className: "radiobtn-content",
    error: error,
    theme: theme,
    tabIndex: -1,
    border: border,
    size: size,
    disabled: disabled
  }, label || children));
  return /*#__PURE__*/React.createElement(_RadioButtonStyle.RadioContainer, (0, _extends2["default"])({
    className: (0, _classnames["default"])('aries-radiobtn', className),
    error: error,
    tabIndex: 0,
    theme: theme,
    border: border,
    disabled: disabled
  }, labelProps), /*#__PURE__*/React.createElement("input", (0, _extends2["default"])({
    type: "radio",
    disabled: disabled
  }, inputProps)), border ? /*#__PURE__*/React.createElement(_RadioButtonStyle.Border, {
    error: error,
    disabled: disabled,
    "data-testid": "radio-border"
  }, content) : content);
};

exports.RadioButton = RadioButton;
var _default = RadioButton;
exports["default"] = _default;