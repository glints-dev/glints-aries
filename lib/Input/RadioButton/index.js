"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

exports.__esModule = true;
exports["default"] = void 0;

var _RadioButton = _interopRequireDefault(require("./RadioButton"));

exports.RadioButton = _RadioButton["default"];
var _default = _RadioButton["default"];
exports["default"] = _default;