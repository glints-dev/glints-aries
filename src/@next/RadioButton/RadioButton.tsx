import React, { HtmlHTMLAttributes } from 'react';
import {
  LabelWrapper,
  RadioButtonInput,
  RadioButtonWrapper,
} from './RadioButtonStyle';
import { Typography } from '../Typography';
import { Colors } from '..';

export interface RadioButtonProps
  extends Omit<React.HTMLAttributes<HTMLInputElement>, 'onChange'> {
  label?: string;
  helperText?: string;
  disabled?: boolean;
  name?: string;
  checked?: boolean;
  value?: string;
  onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
}

/**
 * This is a controlled component, so you need to pass in the `checked` and `onChange` prop to make it work.
 */
export const RadioButton = React.forwardRef<HTMLInputElement, RadioButtonProps>(
  function RadioButton(
    {
      label,
      helperText,
      disabled,
      name,
      checked,
      value,
      onChange,
      ...props
    }: RadioButtonProps,
    ref
  ) {
    return (
      <RadioButtonWrapper>
        <RadioButtonInput
          ref={ref}
          disabled={disabled}
          name={name}
          checked={checked}
          value={value}
          type="radio"
          onChange={onChange}
          onMouseDown={e => e.preventDefault()}
          {...props}
        />
        <LabelWrapper data-disabled={disabled}>
          <Typography variant="body1" as="span" color={Colors.Neutral.B18}>
            {label}
          </Typography>
          <Typography variant="subtitle2" as="span" color={Colors.Neutral.B40}>
            {helperText}
          </Typography>
        </LabelWrapper>
      </RadioButtonWrapper>
    );
  }
);
