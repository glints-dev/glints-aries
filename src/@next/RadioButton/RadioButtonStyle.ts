import styled from 'styled-components';
import { Breakpoints, Colors } from '..';
import { space4, space8 } from '../utilities/spacing';
import { borderRadiusHalf } from '../utilities/borderRadius';

export const RadioButtonWrapper = styled.label`
  display: flex;
  align-items: flex-start;
  gap: ${space8};
  padding: ${space4} 0;
`;

export const RadioButtonInput = styled.input`
  cursor: pointer;
  appearance: none;
  margin: 0;
  width: 18px;
  height: 18px;
  border: 2px solid ${Colors.Neutral.B68};
  border-radius: ${borderRadiusHalf};
  transform: translateY(1.5px);

  &::after {
    content: '';
    display: block;
    border-radius: ${borderRadiusHalf};
    width: 8px;
    height: 8px;
    margin: 3px;
  }

  @media (min-width: ${Breakpoints.large}) {
    width: 22px;
    height: 22px;
    transform: none;
    &::after {
      width: 10px;
      height: 10px;
      margin: ${space4};
    }
  }

  &:hover {
    border-color: ${Colors.Neutral.B68};
  }

  &:focus-visible {
    box-shadow: 0px 0px 0px 1px ${Colors.Neutral.B100},
      0px 0px 0px 3px ${Colors.Blue.S54};
    outline: none;
  }

  &:disabled {
    cursor: not-allowed;
    background-color: ${Colors.Neutral.B95};
    border-color: ${Colors.Neutral.B85};
  }

  &:checked {
    border-color: ${Colors.Blue.S99};

    &::after {
      background-color: ${Colors.Blue.S99};
    }

    &:disabled {
      background-color: ${Colors.Neutral.B95};
      border-color: ${Colors.Neutral.B85};

      &::after {
        background-color: ${Colors.Neutral.B85};
      }
    }
  }
`;

export const LabelWrapper = styled.div`
  cursor: pointer;
  display: flex;
  flex-direction: column;
  gap: ${space4};

  &[data-disabled='true'] {
    span {
      color: ${Colors.Neutral.B85};
    }
    cursor: not-allowed;
  }
`;
