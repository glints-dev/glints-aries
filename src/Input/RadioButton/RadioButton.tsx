import * as React from 'react';

import classNames from 'classnames';

import {
  RadioContainer,
  RadioLabel,
  RadioIcon,
  Border,
} from '../../Style/Input/RadioButtonStyle';

const RadioButton: React.FunctionComponent<Props> = props => {
  const {
    className,
    children,
    error,
    label,
    labelProps,
    theme,
    border,
    disabled,
    size = 'regular',
    ...inputProps
  } = props;

  const content = (
    <>
      <RadioIcon
        error={error}
        theme={theme}
        border={border}
        size={size}
        disabled={disabled}
        data-testid="radio-icon"
      />
      <RadioLabel
        className="radiobtn-content"
        error={error}
        theme={theme}
        tabIndex={-1}
        border={border}
        size={size}
        disabled={disabled}
      >
        {label || children}
      </RadioLabel>
    </>
  );

  return (
    <RadioContainer
      className={classNames('aries-radiobtn', className)}
      error={error}
      tabIndex={0}
      theme={theme}
      border={border}
      disabled={disabled}
      {...labelProps}
    >
      <input type="radio" disabled={disabled} {...inputProps} />
      {border ? (
        <Border error={error} disabled={disabled} data-testid="radio-border">
          {content}
        </Border>
      ) : (
        content
      )}
    </RadioContainer>
  );
};

type HTMLInputProps = Omit<React.HTMLProps<HTMLInputElement>, 'size' | 'label'>;

export interface Props extends HTMLInputProps {
  error?: boolean;
  label?: React.ReactNode;
  labelProps?: React.LabelHTMLAttributes<{}>;
  theme?: 'white';
  border?: boolean;
  disabled?: boolean;
  size?: 'regular' | 'small';
}

export default RadioButton;
