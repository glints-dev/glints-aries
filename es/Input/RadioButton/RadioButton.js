import _extends from "@babel/runtime/helpers/extends";
import _objectWithoutPropertiesLoose from "@babel/runtime/helpers/objectWithoutPropertiesLoose";
var _excluded = ["className", "children", "error", "label", "labelProps", "theme", "border", "disabled", "size"];
import * as React from 'react';
import classNames from 'classnames';
import { RadioContainer, RadioLabel, RadioIcon, Border } from './RadioButtonStyle';
/** The className will be passed to
 * the main radio button container. ...inputProps is passed to the
 * input tag. */

export var RadioButton = function RadioButton(_ref) {
  var className = _ref.className,
      children = _ref.children,
      _ref$error = _ref.error,
      error = _ref$error === void 0 ? false : _ref$error,
      label = _ref.label,
      labelProps = _ref.labelProps,
      theme = _ref.theme,
      _ref$border = _ref.border,
      border = _ref$border === void 0 ? false : _ref$border,
      _ref$disabled = _ref.disabled,
      disabled = _ref$disabled === void 0 ? false : _ref$disabled,
      _ref$size = _ref.size,
      size = _ref$size === void 0 ? 'regular' : _ref$size,
      inputProps = _objectWithoutPropertiesLoose(_ref, _excluded);

  var content = /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(RadioIcon, {
    error: error,
    theme: theme,
    border: border,
    size: size,
    disabled: disabled,
    "data-testid": "radio-icon"
  }), /*#__PURE__*/React.createElement(RadioLabel, {
    className: "radiobtn-content",
    error: error,
    theme: theme,
    tabIndex: -1,
    border: border,
    size: size,
    disabled: disabled
  }, label || children));
  return /*#__PURE__*/React.createElement(RadioContainer, _extends({
    className: classNames('aries-radiobtn', className),
    error: error,
    tabIndex: 0,
    theme: theme,
    border: border,
    disabled: disabled
  }, labelProps), /*#__PURE__*/React.createElement("input", _extends({
    type: "radio",
    disabled: disabled
  }, inputProps)), border ? /*#__PURE__*/React.createElement(Border, {
    error: error,
    disabled: disabled,
    "data-testid": "radio-border"
  }, content) : content);
};
export default RadioButton;