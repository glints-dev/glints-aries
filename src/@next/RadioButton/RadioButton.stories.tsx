import React, { useState } from 'react';

import { Story, Meta } from '@storybook/react';

import { BaseContainer } from '../../Layout/GlintsContainer/GlintsContainer';
import { RadioButton, RadioButtonProps } from './RadioButton';
import { Typography } from '../Typography';

(RadioButton as React.FunctionComponent<RadioButtonProps>).displayName =
  'RadioButton';

export default {
  title: '@next/RadioButton',
  component: RadioButton,
  decorators: [Story => <BaseContainer>{Story()}</BaseContainer>],
} as Meta;

const Template: Story = args => {
  return <RadioButton {...args} />;
};

export const Interactive = Template.bind({});
Interactive.args = {
  label: 'Label',
  helperText: 'Helper Text',
  disabled: false,
  name: 'name',
  checked: false,
  value: 'value',
};

export const ControlledRadioButton = () => {
  const [value, setValue] = useState('1');

  return (
    <>
      <Typography variant="body1">Selected value: {value}</Typography>
      <RadioButton
        name="radio-name"
        label="Label 1"
        value="1"
        checked={value === '1'}
        onChange={e => setValue(e.target.value)}
      />
      <RadioButton
        name="radio-name"
        label="Label 2"
        value="2"
        checked={value === '2'}
        onChange={e => setValue(e.target.value)}
      />
      <RadioButton
        name="radio-name"
        label="Label 3"
        value="3"
        checked={value === '3'}
        onChange={e => setValue(e.target.value)}
      />
    </>
  );
};
