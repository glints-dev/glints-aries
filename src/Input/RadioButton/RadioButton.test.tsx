import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { fireEvent, render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import userEvent from '@testing-library/user-event';

import { PrimaryColor, SecondaryColor, Greyscale } from '../../Style/Colors';
import RadioButton, { Props } from './RadioButton';

const singleRadioButton = (
  <RadioButton label="Full Time" name="job-type" value="full-time" />
);
const singleRadioButtonWhiteTheme = (
  <RadioButton
    label="Full Time"
    name="job-type"
    value="full-time"
    theme="white"
  />
);
const twoRadioButtons = (
  <React.Fragment>
    <RadioButton label="Full Time" name="job-type" value="full-time" />
    <RadioButton label="Intership" name="job-type" value="intership" />
  </React.Fragment>
);
const twoRadioButtonsOneChecked = (
  <React.Fragment>
    <RadioButton
      label="Full Time"
      name="job-type"
      value="full-time"
      checked={true}
    />
    <RadioButton label="Intership" name="job-type" value="intership" />
  </React.Fragment>
);

const setupRadioButtons = () => {
  const onChange = jest.fn(e => e.persist());
  const { getByTestId } = render(
    <React.Fragment>
      <RadioButton
        label="Full Time"
        name="job-type"
        value="full-time"
        data-testid="radio-1"
        onChange={onChange}
      />
      <RadioButton
        label="Intership"
        name="job-type"
        value="internship"
        data-testid="radio-2"
        onChange={onChange}
      />
    </React.Fragment>
  );
  return {
    onChange,
    radioButton1: getByTestId('radio-1'),
    radioButton2: getByTestId('radio-2'),
  };
};

describe('<RadioButton>', () => {
  it('should render as expected', () => {
    const singleRadioSnapshot = renderer.create(singleRadioButton).toJSON();
    expect(singleRadioSnapshot).toMatchSnapshot();
    const singleRadioSnapshotWhiteTheme = renderer
      .create(singleRadioButtonWhiteTheme)
      .toJSON();
    expect(singleRadioSnapshotWhiteTheme).toMatchSnapshot();
    const twoRadiosSnapshot = renderer.create(twoRadioButtons).toJSON();
    expect(twoRadiosSnapshot).toMatchSnapshot();
    const twoRadioButtonsOneCheckedSnapshot = renderer
      .create(twoRadioButtonsOneChecked)
      .toJSON();
    expect(twoRadioButtonsOneCheckedSnapshot).toMatchSnapshot();
  });

  it('should call onChange when clicked', () => {
    const { onChange, radioButton1, radioButton2 } = setupRadioButtons();
    fireEvent.click(radioButton1);
    expect(onChange).toHaveBeenCalledTimes(1);
    expect(onChange.mock.calls[0][0].target.value).toBe('full-time');
    fireEvent.click(radioButton2);
    expect(onChange).toHaveBeenCalledTimes(2);
    expect(onChange.mock.calls[1][0].target.value).toBe('internship');
  });

  it('should not call onChange again when clicked a second time', () => {
    const { onChange, radioButton1 } = setupRadioButtons();
    fireEvent.click(radioButton1);
    fireEvent.click(radioButton1);
    expect(onChange).toHaveBeenCalledTimes(1);
  });

  it('should should check the clicked item', () => {
    const { radioButton1, radioButton2 } = setupRadioButtons();
    fireEvent.click(radioButton1);
    expect(radioButton1).toBeChecked();
    fireEvent.click(radioButton2);
    expect(radioButton1).not.toBeChecked();
    expect(radioButton2).toBeChecked();
  });

  it('should not call onChange again when disabled', () => {
    const onChange = jest.fn();
    const { getByTestId } = render(
      <RadioButton
        label="Full Time"
        name="job-type"
        value="full-time"
        data-testid="radio"
        onChange={onChange}
        disabled={true}
      />
    );
    const radioButton = getByTestId('radio');
    userEvent.click(radioButton);
    expect(onChange).not.toHaveBeenCalled();
  });
});

describe('<RadioButton /> color', () => {
  const onChange = jest.fn();
  const Component = (props: Props) => (
    <RadioButton
      label="Full Time"
      name="job-type"
      value="full-time"
      onChange={onChange}
      {...props}
    />
  );
  it('should be grey by default', () => {
    const { getByTestId } = render(<Component />);
    const radioIcon = getByTestId('radio-icon');
    expect(radioIcon).toHaveStyle(`border-color: ${Greyscale.grey}`);
  });

  it('should be green when checked', () => {
    const { getByTestId } = render(<Component checked />);
    const radioIcon = getByTestId('radio-icon');
    expect(radioIcon).toHaveStyle(`border-color: ${SecondaryColor.darkgreen}`);
  });

  it('should be red when has error', () => {
    const { getByTestId } = render(<Component error />);
    const radioIcon = getByTestId('radio-icon');
    expect(radioIcon).toHaveStyle(`border-color: ${PrimaryColor.glintsred}`);
  });

  it('should be light grey when disabled', () => {
    const { getByTestId } = render(<Component disabled />);
    const radioIcon = getByTestId('radio-icon');
    expect(radioIcon).toHaveStyle(`border-color: ${Greyscale.lightgrey}`);
  });
});

describe('<RadioButton border /> color', () => {
  const onChange = jest.fn();
  const Component = (props: Props) => (
    <RadioButton
      label="Full Time"
      name="job-type"
      value="full-time"
      onChange={onChange}
      border={true}
      {...props}
    />
  );
  it('should be grey by default', () => {
    const { getByTestId } = render(<Component />);
    const radioIcon = getByTestId('radio-icon');
    const radioBorder = getByTestId('radio-border');
    expect(radioIcon).toHaveStyle(`border-color: ${Greyscale.grey}`);
    expect(radioBorder).toHaveStyle('border-color: #aaa');
  });

  it('should be blue when checked', () => {
    const { getByTestId } = render(<Component checked />);
    const radioIcon = getByTestId('radio-icon');
    const radioBorder = getByTestId('radio-border');
    expect(radioIcon).toHaveStyle(`border-color: ${SecondaryColor.actionblue}`);
    expect(radioBorder).toHaveStyle(
      `border-color: ${SecondaryColor.actionblue}`
    );
  });

  it('should be blue when error', () => {
    const { getByTestId } = render(<Component error />);
    const radioIcon = getByTestId('radio-icon');
    const radioBorder = getByTestId('radio-border');
    expect(radioIcon).toHaveStyle(`border-color: ${Greyscale.grey}`);
    expect(radioBorder).toHaveStyle(`border-color: ${PrimaryColor.glintsred}`);
    expect(radioBorder).toHaveStyle('background-color: rgba(236, 39, 43, 0.1)');
  });

  it('should be light grey when disabled', () => {
    const { getByTestId } = render(<Component disabled />);
    const radioIcon = getByTestId('radio-icon');
    const radioBorder = getByTestId('radio-border');
    expect(radioIcon).toHaveStyle(`border-color: ${Greyscale.lightgrey}`);
    expect(radioBorder).toHaveStyle(`border-color: ${Greyscale.lightgrey}`);
  });
});
