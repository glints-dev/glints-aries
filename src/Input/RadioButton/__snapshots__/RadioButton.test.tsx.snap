// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioButton> should render as expected 1`] = `
<label
  className="RadioButtonStyle__RadioContainer-sc-1xclcfi-3 eyRelA aries-radiobtn"
  tabIndex={0}
>
  <input
    name="job-type"
    type="radio"
    value="full-time"
  />
  <span
    className="RadioButtonStyle__RadioIcon-sc-1xclcfi-0 hinYyw"
    data-testid="radio-icon"
    size="regular"
  />
  <span
    className="RadioButtonStyle__RadioLabel-sc-1xclcfi-1 xviKd radiobtn-content"
    size="regular"
    tabIndex={-1}
  >
    Full Time
  </span>
</label>
`;

exports[`<RadioButton> should render as expected 2`] = `
<label
  className="RadioButtonStyle__RadioContainer-sc-1xclcfi-3 kINzCj aries-radiobtn"
  tabIndex={0}
>
  <input
    name="job-type"
    type="radio"
    value="full-time"
  />
  <span
    className="RadioButtonStyle__RadioIcon-sc-1xclcfi-0 fYRZeZ"
    data-testid="radio-icon"
    size="regular"
  />
  <span
    className="RadioButtonStyle__RadioLabel-sc-1xclcfi-1 ioQzrz radiobtn-content"
    size="regular"
    tabIndex={-1}
  >
    Full Time
  </span>
</label>
`;

exports[`<RadioButton> should render as expected 3`] = `
Array [
  <label
    className="RadioButtonStyle__RadioContainer-sc-1xclcfi-3 eyRelA aries-radiobtn"
    tabIndex={0}
  >
    <input
      name="job-type"
      type="radio"
      value="full-time"
    />
    <span
      className="RadioButtonStyle__RadioIcon-sc-1xclcfi-0 hinYyw"
      data-testid="radio-icon"
      size="regular"
    />
    <span
      className="RadioButtonStyle__RadioLabel-sc-1xclcfi-1 xviKd radiobtn-content"
      size="regular"
      tabIndex={-1}
    >
      Full Time
    </span>
  </label>,
  <label
    className="RadioButtonStyle__RadioContainer-sc-1xclcfi-3 eyRelA aries-radiobtn"
    tabIndex={0}
  >
    <input
      name="job-type"
      type="radio"
      value="intership"
    />
    <span
      className="RadioButtonStyle__RadioIcon-sc-1xclcfi-0 hinYyw"
      data-testid="radio-icon"
      size="regular"
    />
    <span
      className="RadioButtonStyle__RadioLabel-sc-1xclcfi-1 xviKd radiobtn-content"
      size="regular"
      tabIndex={-1}
    >
      Intership
    </span>
  </label>,
]
`;

exports[`<RadioButton> should render as expected 4`] = `
Array [
  <label
    className="RadioButtonStyle__RadioContainer-sc-1xclcfi-3 eyRelA aries-radiobtn"
    tabIndex={0}
  >
    <input
      checked={true}
      name="job-type"
      type="radio"
      value="full-time"
    />
    <span
      className="RadioButtonStyle__RadioIcon-sc-1xclcfi-0 hinYyw"
      data-testid="radio-icon"
      size="regular"
    />
    <span
      className="RadioButtonStyle__RadioLabel-sc-1xclcfi-1 xviKd radiobtn-content"
      size="regular"
      tabIndex={-1}
    >
      Full Time
    </span>
  </label>,
  <label
    className="RadioButtonStyle__RadioContainer-sc-1xclcfi-3 eyRelA aries-radiobtn"
    tabIndex={0}
  >
    <input
      name="job-type"
      type="radio"
      value="intership"
    />
    <span
      className="RadioButtonStyle__RadioIcon-sc-1xclcfi-0 hinYyw"
      data-testid="radio-icon"
      size="regular"
    />
    <span
      className="RadioButtonStyle__RadioLabel-sc-1xclcfi-1 xviKd radiobtn-content"
      size="regular"
      tabIndex={-1}
    >
      Intership
    </span>
  </label>,
]
`;
